falcon/media/multipart.py

```diff
@@ -592,6 +592,15 @@ class MultipartParseOptions:
     If the body part headers size exceeds this value, an instance of
     :class:`.MultipartParseError` will be raised.
     """
+    max_filename_length: Optional[int]
+    """The maximum length of filenames in multipart forms (default ``None``).
+
+    If the filename exceeds this value, it will be truncated while preserving
+    the file extension. Defaults to ``None`` for backward compatibility.
+
+    .. note::
+        Starting from Falcon 5.0, this will default to 255 characters.
```

Comment:
I suggest a smaller default value. On some systems, e.g., Windows, the max possible path length is around that value, so likely we want to make the file name part of the path shorter.

Maybe 64 could be a sane default?

Reply:
64 may be a bit small as default, but 96/128 is likely ok.

Reply:
Personally I think 64 ought to be enough for the basename, since `secure_filename()` is mangling the name in any case, so there is no guarantee to get exactly the same string as submitted in the form anyway. But 96 works for me as well.